shared/utils/polyvinyl.ts

```diff
@@ -103,16 +103,17 @@ function checkPoly(poly: ChallengeFile) {
   );
 }
 
-// setContent will lose source if set
+// setContent will lose source if not supplied
 export function setContent(
   contents: string,
-  poly: ChallengeFile
+  poly: ChallengeFile,
+  source?: string | null
 ): ChallengeFile {
   checkPoly(poly);
   return {
     ...poly,
     contents,
-    source: null
+    source: source ?? null
```

Comment:
```suggestion
    source
```

We don't need nullish coalescing for a `string | null` value if the fallback is also `null`.

Reply:
Good point. Now that you mention it, we can get rid of the `null`s entirely.